vyper/compiler/output.py

```diff
@@ -35,11 +33,47 @@ def build_userdoc(compiler_data: CompilerData) -> dict:
 
 
 def build_external_interface_output(compiler_data: CompilerData) -> str:
-    return extract_external_interface(compiler_data.global_ctx, compiler_data.contract_name)
+    interface = compiler_data.vyper_module_folded._metadata["type"]
+    name = Path(compiler_data.contract_name).stem.capitalize()
+    out = f"\n# External Interfaces\ninterface {name}:\n"
+
+    for func in interface.members.values():
+        if func.visibility == FunctionVisibility.INTERNAL or func.name == "__init__":
+            continue
+        args = ", ".join([f"{name}: {typ}" for name, typ in func.arguments.items()])
+        return_value = f" -> {func.return_type}" if func.return_type is not None else ""
+        mutability = func.mutability.value
+        out = f"{out}    def {func.name}({args}){return_value}: {mutability}\n"
+
+    return out
 
 
 def build_interface_output(compiler_data: CompilerData) -> str:
-    return extract_interface_str(compiler_data.global_ctx)
+    interface = compiler_data.vyper_module_folded._metadata["type"]
+    out = ""
+
+    if interface.events:
+        out = "# Events\n\n"
+        for event in interface.events.values():
+            out = f"{out}event {event.name}:\n"
```

Comment:
why not `out += f"event {event.name}:\n"`?

Reply:
https://softwareengineering.stackexchange.com/questions/90203/why-is-so-bad-for-concatenation

Reply:
hmm, I wonder if this could be refactored using `str.join(*iterable*)`?

https://stackoverflow.com/questions/1316887/what-is-the-most-efficient-string-concatenation-method-in-python

Reply:
maybe:
```python
out = "# Events\n\n" + \
    "\n\n".join(
        "".join(
            f"event {event.name}:\n    ",
            (
            "\n    ".join(
                f"{name}: {typ}"
                for name, typ in event.arguments.items()
            )
        )
        if event.arguments
        else "pass"
        for event in interface.events.values()
    )

```

Reply:
```python
        out = "# Events\n\n"
        for event in interface.events.values():
            encoded_args = "\n    ".join(f"{name}: {typ}" for name, typ in event.arguments.items())
            out = f"{out}event {event.name}:\n    {encoded_args if event.arguments else 'pass'}\n"
```